src/EffectManager.cpp

```diff
@@ -135,3 +158,58 @@ ChainCallbackResult EffectManager::OnDraw(EffectManager *mgr)
     }
     return CHAIN_CALLBACK_RESULT_CONTINUE;
 }
+
+#pragma var_order(effect, idx)
+Effect *EffectManager::SpawnParticles(i32 effectIdx, D3DXVECTOR3 *pos, i32 count, ZunColor color)
+{
+    i32 idx;
+    Effect *effect;
+
+    effect = &this->effects[this->nextIndex];
+    for (idx = 0; idx < ARRAY_SIZE_SIGNED(this->effects); idx++)
+    {
+        this->nextIndex++;
+        if (this->nextIndex >= ARRAY_SIZE_SIGNED(this->effects))
+        {
+            this->nextIndex = 0;
+        }
+        if (effect->inUseFlag)
+        {
+            if (!this->nextIndex)
```

Comment:
```suggestion
            if (this->nextIndex == 0) {
                effect = &this->effects[0]
            }
```

I think this is a bit easier to understand what's going on.